gulpfile.js

```diff
@@ -0,0 +1,101 @@
+var minifyCSS = require('gulp-minify-css');
+var ifElse = require('gulp-if-else');
+var argv = require('yargs')
+    .usage('Usage: $0 --appengine [Path to app engine]' +
+      '--clear_datastore [clear all data storage]' +
+      '--enable_sendmail [send emails]' +
+      '--dev_mode [variable that shows if is in dev_mode]')
+    .demand(['appengine']).argv;
+var manifest = require('./manifest.json');
+
+var serveCmd = argv.appengine;
+var params = {
+  admin_host: '0.0.0.0',
+  admin_port: 8000,
+  host: '0.0.0.0',
+  port: 8181
+};
+
+var is_dev = argv.dev_mode == 'True';
+var dependencies = manifest.dependencies;
+var css_files = [];
+var font_folders = [];
+var css_background = [];
+for (var obj in dependencies) {
+  var dependency = dependencies[obj];
+  if (dependency.hasOwnProperty('cssFiles')) {
+    var cssFiles = dependency['cssFiles'];
+    for (var files in cssFiles) {
+      css_files.push(path.join('third_party', 'static',
+      dependency['dest_target'] + dependency['version'],
+      cssFiles[files]));
+    }
+  }
+  if (dependency.hasOwnProperty('fonts_path')) {
+    var font_prefix = '*.{eot,woff2,ttf,woff,eof,svg}';
+    font_folders.push(path.join('third_party', 'static',
+      dependency['dest_target'] + dependency['version'],
+      dependency['fonts_path'], font_prefix));
+  }
+  if (dependency.hasOwnProperty('css_background_image')) {
+    var background_image = dependency['css_background_image'];
+    for (var image in background_image) {
+      css_background.push(path.join('third_party', 'static',
+        dependency['dest_target'] + dependency['version'],
+        background_image[image]));
+    }
+  }
+}
+
+if (argv.clear_datastore) {
+  params.clear_datastore = true;
+}
+
+if (argv.enable_sendmail) {
+  params.enable_sendmail = argv.enable_sendmail;
+}
+
+var css_path = path.join('core', 'templates', is_dev ? 'dev' : 'prod', 'head', 'css');
+gulp.task('css', function() {
+  gulp.src(css_files)
+    .pipe(ifElse(!is_dev, minifyCSS))
+    .pipe(concatCSS('third_party.css'))
+    .pipe(gulp.dest(css_path));
+});
+
+gulp.task('copyFonts', function() {
+  gulp.src(font_folders)
+  .pipe(gulp.dest(path.join('static', 'fonts')));
+});
+
+gulp.task('copyCssBackgroundImages', function() {
+  gulp.src(css_background)
+  .pipe(gulp.dest(css_path));
+});
+
+gulp.task('gulp-serve', function() {
+  gulp.src('app.yaml')
+    .pipe(oppiaGae(serveCmd, [], params));
+});
+
+// This takes all functions  that are required for the build
+// e.g css, Js and Images
+gulp.task('build', ['css', 'copyFonts', 'copyCssBackgroundImages']);
+
+gulp.slurped = false;
+gulp.task('watch', function() {
+  if (!gulp.slurped) {
+    gulp.watch('gulpfile.js', ['build']);
+    gulp.watch(css_files, ['css']);
+    gulp.watch('manifest.json', ['build']);
+    gulp.slurped = true;
+  }
+});
+
+// This takes all default task which must be run when start.sh is started.
+gulp.task('.', ['build', 'gulp-serve', 'watch']);
+
```

Comment:
As we discussed this gulp will also be run by scripts/build.py but this will also minify css for production purpose.